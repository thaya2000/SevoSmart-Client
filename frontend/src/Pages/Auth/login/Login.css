



.logincontainer{
    display: flex;
    flex-direction: column;
    background: #fff;
    margin: auto;
    width:100%;
    height:80%;
    border-radius: 15px;
   

}

.header{
    display: flex;
    flex-direction: column; 
    align-items: flex-start; 
    margin-top: 30px;
}

.text{
    color:black;
    font-size: 40px;
    font-weight:bolder;
}


.inputs{
    margin-top: 30px;
    display: flex;
    flex-direction: column;
    gap: 25px;
    align-items: flex-start;
    width: 100%;
}
.input-container{
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: auto;
    width: 100%;
    height: 40px;
    background:#D9D9D9;
   
    gap: 10px;
}

.input-container input{
 
    background: transparent;
    border: none;
    outline: none;
    color: black;
    font-size: 15px;
    margin:auto;
    padding: auto;
    display: flex;
}
.user-name-input{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    font-weight: bolder;
}
.password-input{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    font-weight: bolder;
}

.submit-container{
    display: flex;
    flex-direction: column;
    gap: 10px;
    margin: 60px auto;
    width: 100%;

}

.submit{
    display: flex;
    justify-content: center;
    align-items: center;
    width: 220px;
    height: 59px;
    color: #fff;
    background: #4c00b4;
    border-radius: 50px;
    font-size: 20px;
    font-weight: 700;
    cursor: pointer;
}

.login-button{
   
    background-color:#7E9AFF;
    height: 35px;
    font-weight: bolder;
  
    
}
.create-account-button{
    background-color: #D9D9D9;
    height: 35px;
    font-weight: bolder;
}

.underline-container{
    display: flex;
    flex-direction: row;
    align-items:center;
}
.or-text{
    padding:5px;
    margin: 0 5px;
}

.underline{
    width:100%;
    height: 1px;
    background: black;
    border-radius: 9px;
  
}

  
  








