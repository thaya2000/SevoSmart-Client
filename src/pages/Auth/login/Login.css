.login-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 25rem;
  height: 30rem;
  background-color: rgb(255, 255, 255);
  position: relative;
  border-radius: 2rem;
  font-weight: 600;
  font-size: 2rem;
  
  border:2px solid rgb(193, 189, 189);
  backdrop-filter: blur(5px);
}



.login-input {
  display: block;
  width: 300px; /* Adjust width to accommodate the icon */
  height: 2.7rem;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 1rem;
  padding: 1rem;
  font-weight: 300;
  font-size: 1rem;
  background-color: rgb(243, 236, 236);
  border: 2px solid rgba(4, 4, 4, 0.2);
}




.user-icon{
  position:absolute;
  right: 20px;
  top:50%;
  transform:translateY(-50%);
  font-size: 16px;
  color: rgb(122, 124, 124);
  
}
.lock-icon{
  position:absolute;
  right: 20px;
  top:50%;
  transform:translateY(-50%);
  font-size: 16px;
  color: rgb(122, 124, 124);
  
}


.user-login{
  width:100%;
  height:100px;
  background-color: grey;
  margin: 30px 0;

  margin-top: -200px;
}
.user-name-login{
  font-family: 'Courier New', Courier, monospace;
  font-size: 20px;
  margin-left: 18px;
}


.create-account{
  margin-top: -70px;
  margin-bottom: 10px;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
color: rgb(122, 124, 124);
  text-decoration-skip-ink: none; /* Ensure underline covers entire text */
  padding-bottom: 5px; /* Increase space beneath the text */
  border-bottom: 3px solid rgb(83, 85, 86);
}
.user-input-container{

  position:relative;
  margin-top: 50px;
  margin-bottom: 40px;
}

.password-input-container{
  position: relative;
}

.submit-button-login {
  top:80%;
  width: 100px; /* Adjust width as needed */
  height: 35px; /* Adjust height as needed */
  color: rgb(148, 144, 144); /* Set text color */
  border: 2px solid rgb(149, 143, 143); /* Remove border */
  border-radius: 20px; /* Make it rounded */
  display: flex; /* Use flexbox */
  justify-content: center; /* Center horizontally */
  align-items: center; /* Center vertically */
  cursor: pointer; /* Add cursor pointer for interaction */
  position: absolute; /* Position absolute */
 font-size: 18px;
  left: 50%; /* Center horizontally */
  transform: translate(-50%, -50%); /* Centering trick */
  background-color: blue;
  /* Hover effect */
  transition: background-color 0.3s ease; /* Smooth transition */
}

/* Hover effect */
.submit-button-login:hover {
  background-color: rgb(5, 45, 126); /* Change background color on hover */
}
