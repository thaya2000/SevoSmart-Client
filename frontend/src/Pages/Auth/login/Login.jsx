import React from 'react'
import { UilTimes, UilUser } from "@iconscout/react-unicons";
import logo from "../../../images/logo.png";
import { Link } from "react-router-dom";
import "./Login.css";
import ConstructionsBg from '../../../images/ConstructionsBg.jpg';


const Login = () => {
  return (

    <div className="logincontainer">
      <div className="login-header">
          <img src={logo} alt="" />
          <Link to="/">
              <button className='login-close-button'><UilTimes /></button>
          </Link>
      </div>
      <div className="login-Card" style={{ backgroundImage: `url(${ConstructionsBg})` }}>
          <div className="login-card" >
            <span className='signin'>Sign In</span>
            <div className="inputs">
            <div className="user-name-input">
              <div>
                <text>User Name</text>
              </div>
              <div className="input-container">
                <input type="text" />
              </div>
            </div>

            <div className="password-input">
              <div>
                <text>Password</text>
              </div>

              <div className="input-container">
                <input type="password" />
              </div>
            </div>
        </div>
        <div className="submit-container">
            <button className="login-button">Sign In</button>
            <div className="underline-container">
              <div className="underline"></div>
              <div className="or-text">
                <text>Or</text>
              </div>
              <div className="underline"></div>
            </div>
            {/* <Link to='/signup'> */}
              <button className="create-account-button">Create Account</button>
            {/* </Link> */}
            
        </div>
      </div>
      </div>
      

          

          
    </div>

  );
};

export default Login