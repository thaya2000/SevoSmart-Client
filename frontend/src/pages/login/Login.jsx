import React from "react";
import "./Login.css";
import { AcmeLogo } from "../../components/nav/AcmeLogo";

export const Login = () => {
  return (
    <div>
      <AcmeLogo />

      <div className="login-background">
        <div className="logincontainer">
          <div className="header">
            <div className="text">Sign In</div>
          </div>

          <div className="inputs">
            <div className="user-name-input">
              <div>
                <text>User Name</text>
              </div>
              <div className="input-container">
                <input className="username" type="text" />
              </div>
            </div>

            <div className="password-input">
              <div>
                <text>Password</text>
              </div>

              <div className="input-container">
                <input type="password" />
              </div>
            </div>
          </div>

          <div className="submit-container">
            <button className="login-button">Log In</button>
            <div className="underline-container">
              <div className="underline"></div>
              <div className="or-text">
                <text>Or</text>
              </div>
              <div className="underline"></div>
            </div>

            <button className="create-account-button">Create Account</button>
          </div>
        </div>
      </div>
    </div>
  );
};
