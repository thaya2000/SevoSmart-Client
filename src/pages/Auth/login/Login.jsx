import "./Login.css";
import { useState } from "react";
import { userAuth } from "../../../context/authContext";
import { useNavigate, useLocation } from "react-router-dom";
import axios from "axios";
import toast from "react-hot-toast";
import { FaUser, FaLock } from "react-icons/fa";
const Login = () => {
  const [email, setEmail] = useState("user@gmail.com");
  const [password, setPassword] = useState("user1234");

  // hook
  const [auth, setAuth] = userAuth();
  const navigate = useNavigate();
  const location = useLocation();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const { data } = await axios.post(`/authenticate`, {
        email,
        password,
      });
      console.log(data);
      if (data?.message) {
        toast.error(data.message);
      } else {
        localStorage.setItem("auth", JSON.stringify(data));
        setAuth({ ...auth, token: data.token, user: data.user });
        navigate(
          location.state || `/`
          // `/dashboard/${data?.user?.role === 1 ? "admin" : "user"}`
        );
        toast.success("Login successful");
      }
    } catch (err) {
      console.log(err);
      toast.error("Login failed. Try again.");
    }
  };
  return (
    <div className="flex relative items-center justify-center w-100v h-100v">
      <div className="login-container">
        <div className="create-account">Login</div>
        <form onSubmit={handleSubmit} className="from-login">
          <div className="user-input-container">
            <input
              type="text"
              className="login-input"
              placeholder="Enter your email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
            />
            <FaUser className="user-icon" />
          </div>
          <div className="password-input-container">
            <input
              type="text"
              className="login-input"
              placeholder="Enter your password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
            <FaLock className="lock-icon" />
          </div>
          <div
            className="submit-button-login "
            // type="submit"
            onClick={handleSubmit}
          >
            Login
          </div>
        </form>
      </div>
    </div>
  );
};

export default Login;
