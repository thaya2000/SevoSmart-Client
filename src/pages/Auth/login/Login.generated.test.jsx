import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import Login from './Login';

// Mocking styles and modules
vi.mock("./Login.css");
vi.mock("../../../context/authContext", () => ({
  userAuth: vi.fn(() => [{}]) // Mock userAuth with a function that returns an empty context
}));
vi.mock("react-router-dom", () => ({
  useNavigate: vi.fn(),
  useLocation: vi.fn()
}));
vi.mock("axios");
vi.mock("react-hot-toast");
vi.mock("react-icons/fa", () => ({
  FaUser: vi.fn(() => <div>FaUser Icon</div>),
  FaLock: vi.fn(() => <div>FaLock Icon</div>)
}));
vi.mock("../../../util/constants", () => ({
  DEFAULT_EMAIL: "example@example.com",
  DEFAULT_PASSWORD: "password"
}));

describe('<Login>', () => {
  it('should render component', () => {
    const { asFragment } = render(<Login />);
    expect(asFragment()).toMatchSnapshot();
  });
});
