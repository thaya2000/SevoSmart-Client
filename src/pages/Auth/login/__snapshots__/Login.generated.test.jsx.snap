// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Login> > should render component 1`] = `
<DocumentFragment>
  <div
    class="flex relative items-center justify-center w-100v h-100v"
  >
    <div
      class="login-container"
    >
      <div
        class="create-account"
      >
        Login
      </div>
      <form
        class="from-login"
      >
        <div
          class="user-input-container"
        >
          <input
            class="login-input"
            placeholder="Enter your email"
            type="text"
            value="example@example.com"
          />
          <div>
            FaUser Icon
          </div>
        </div>
        <div
          class="password-input-container"
        >
          <input
            class="login-input"
            placeholder="Enter your password"
            type="text"
            value="password"
          />
          <div>
            FaLock Icon
          </div>
        </div>
        <div
          class="submit-button-login "
        >
          Login
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
